### PR TITLE
doc: Fixing board names in samples, applications and tests

### PR DESCRIPTION
**Pull request by umapraseeda**
Fixing board names in samples,applications and tests
ref: https://projecttools.nordicsemi.no/jira/browse/NCSDK-3933

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

### CODE REVIEW

File: applications/nrf_desktop/README.rst [hunk 1]
```diff
@@ -241,40 +241,47 @@ Depending on what development kit you use, you need to select the respective con
...
 
    .. tab:: DK
 
       .. figure:: /images/nrf_desktop_nrf52840_dk_no_captions.svg
          :alt: DK
 
-      * nRF52840 DK (nrf52840dk_nrf52840) - The application is configured to work as gaming mouse (with motion emulated by using DK buttons).
-      * nRF52833 DK (nrf52833dk_nrf52833) - The application is configured to work as HID dongle.
+      .. include:: /includes/boardname_tables/sample_boardnames.rst
+         :start-after: thread_zigbee_set1_start
```
Comment (pdunaj) [inline applications/nrf_desktop/README.rst:281]: thread_zigbee_set1_start?

why naming generic boards "thread_zigbee"?

File: applications/nrf_desktop/README.rst [hunk 2]
```diff
@@ -241,40 +241,47 @@ Depending on what development kit you use, you need to select the respective con
...
+         :end-before: thread_zigbee_set1_end
 
+      In nRF52840 DK, the application is configured to work as gaming mouse (with motion emulated by using DK buttons) and in nRF52833 DK, the application is configured to work as HID dongle.
```
Comment (pdunaj) [inline applications/nrf_desktop/README.rst:284]: add an empty line after this and before `..`

File: samples/bootloader/README.rst [hunk 3]
```diff
@@ -85,12 +85,11 @@ The bootloader uses the :ref:`doc_bl_storage` library to access provisioned data
...
-* One of the following development boards:
+The sample supports the following development kits:
 
-  * |nRF9160DK|
-  * |nRF52840DK|
-  * |nRF52DK|
-  * |nRF51DK|
+.. include:: /includes/boardname_tables/sample_boardnames.rst
+   :start-after: misc_samples_set1_start
+   :end-before: misc_samples_set1_end
```
Comment (hakonfam) [inline samples/bootloader/README.rst:92]: This set seems to miss nrf51dk?

File: doc/nrf/index.rst [hunk 4]
```diff
@@ -36,3 +36,5 @@ Documentation for different versions of the |NCS| is available at the following
    scripts
 
 ..   cheat_sheet
+
+..   sample_boardnames.rst
```
Comment (ru-fu) [inline doc/nrf/index.rst:40]: This seems to work, so it's okay for now, but ideally we should change the file name to sample_boardnames.txt (like our other include files). Then it won't be copied and included as rst file.

File: samples/bootloader/README.rst (hunk 3 above)
Comment (ru-fu) [reply samples/bootloader/README.rst:92]: We don't officially support nRF51 DK in the NCS.

**Comment by ru-fu** [comment]
We won't manage to get this PR in for the release. We'll make a new PR that updates only those samples that have been reviewed and approved, and the other samples will need to be updated afterwards.

File: samples/usb/usb_uart_bridge/README.rst [hunk 5]
```diff
@@ -13,11 +13,13 @@ The USB-UART bridge acts as a serial adapter, exposing 2 UART pairs to a USB hos
 Requirements
 ************
 
-* One of the following development boards:
+The sample supports the following nRF9160-based device:
```
Comment (jtguggedal) [inline samples/usb/usb_uart_bridge/README.rst:16]: The target is the nRF5280 on Thingy:91

File: samples/usb/usb_uart_bridge/README.rst [hunk 6]
```diff
@@ -13,11 +13,13 @@ The USB-UART bridge acts as a serial adapter, exposing 2 UART pairs to a USB hos
...
 
-  * |Thingy91|
+.. include:: /includes/boardname_tables/sample_boardnames.txt
+   :start-after: set10_start
```
Comment (jtguggedal) [inline samples/usb/usb_uart_bridge/README.rst:19]: If I understand this reference correctly, it should be changed to reflect `thingy91_nrf52840` as build target

File: applications/serial_lte_modem/README.rst [hunk 7]
```diff
@@ -9,32 +9,33 @@ An nRF9160 DK is used as the host, while the client can be simulated using eithe
...
+* Support for FTP client proprietary AT commands
+* Support for communication to external MCU over UART
 
 All nRF91 modem AT commands are also supported.
 
 Requirements
 ************
 
-* The following development board:
+The sample supports the following development kit:
```
Comment (rlubos) [inline applications/serial_lte_modem/README.rst:26]: `serial_lte_modem` was promoted to an `application` rank :)

File: applications/serial_lte_modem/README.rst [hunk 8]
```diff
@@ -9,32 +9,33 @@ An nRF9160 DK is used as the host, while the client can be simulated using eithe
...
 
-    * |nRF9160DK|
+.. include:: /includes/boardname_tables/sample_boardnames.txt
+   :start-after: set5_start
+   :end-before: set5_end
 
-* If the client is a PC:
+If the client is a PC, the sample requires any terminal software, such as TeraTerm.
```
Comment (rlubos) [inline applications/serial_lte_modem/README.rst:32]: Same here.

File: applications/serial_lte_modem/README.rst [hunk 9]
```diff
@@ -9,32 +9,33 @@ An nRF9160 DK is used as the host, while the client can be simulated using eithe
...
 
-    * Any terminal software, such as TeraTerm.
+If the client is an nRF52 device, the sample supports the following development kits:
```
Comment (rlubos) [inline applications/serial_lte_modem/README.rst:34]: And here.

**Comment by carlescufi** [comment]
@umapraseeda can you please fix the build issues so we can merge this?

**Comment by umapraseeda** [comment]
> @umapraseeda can you please fix the build issues so we can merge this?
@carlescufi 
 Instead of merging this PR, we are getting in the changes for specific samples that are approved, through additional small PRs. We can close this PR  after we  finish adding the changes for all the samples and applications through separate PRs.

**Comment by doki-nordic** [comment]
Verified:
samples/nfc/system_off
samples/nrf5340/empty_app_core

**Comment by umapraseeda** [comment]
> doc/nrf/templates/sample_README.txt should be updated as well.

This is being done in https://github.com/nrfconnect/sdk-nrf/pull/2632.

**Comment by joerchan** [comment]
@umapraseeda Please rebase.

File: applications/nrf_desktop/README.rst (hunk 2 above)
Comment (pdunaj) [reply applications/nrf_desktop/README.rst:284]: ping

**Comment by evenl** [comment]
GPS sample changes approved by me.

**Comment by umapraseeda** [comment]
Adding the changes for BLE and NFC samples (Included emulated boards) according to NCSDK-4950 and https://github.com/nrfconnect/sdk-nrf/pull/2731.

File: doc/nrf/includes/boardname_tables/sample_boardnames.txt [hunk 10]
```diff
@@ -1,7 +1,6 @@
 .. _sampleboardnames:
 
-Set used by BLE (Central BAS, Central DFU SMP, Central HIDS, Central UART, EnOcean, Mesh Light, Mesh Light Switch, Peripheral HIDS Keyboard,  Peripheral HIDS mouse,
-Peripheral UART, NUS Shell Transport, Throughput) and NFC samples
+Set used by sample (Radio Test)
```
Comment (joerchan) [inline doc/nrf/includes/boardname_tables/sample_boardnames.txt:3]: Nice, there are atleast 3 open PRs that touch this line, I hope this is the one that will stand

File: doc/nrf/includes/boardname_tables/sample_boardnames.txt [hunk 11]
```diff
@@ -33,27 +32,27 @@ Set used by BLE samples (LLPM, peripheral GATT Discovery Manager), Enhanced Shoc
 
 .. set2_end
 
-Set used by BLE samples (Peripheral LBS)
+Set used by BLE samples (Mesh Light Switch, Throughput, LLPM, NFC Pairing) and NFC samples
```
Comment (joerchan) [inline doc/nrf/includes/boardname_tables/sample_boardnames.txt:35]: BLE -> Bluetooth LE

File: doc/nrf/includes/boardname_tables/sample_boardnames.txt [hunk 12]
```diff
@@ -277,3 +278,105 @@ Set used by the SPM sample
 +--------------------------------+-----------+------------------------------------------------+--------------------------------+
 
 .. set19_end
+
+Set used by BLE samples (Central BAS, Central DFU SMP, Enocean)
```
Comment (joerchan) [inline doc/nrf/includes/boardname_tables/sample_boardnames.txt:282]: BLE -> Bluetooth LE

File: doc/nrf/includes/boardname_tables/sample_boardnames.txt [hunk 13]
```diff
@@ -277,3 +278,105 @@ Set used by the SPM sample
...
++--------------------------------------------------+-------------+------------------------------------------------+--------------------------------+
+|:ref:`nRF52833 DK (emulating nRF52820) <ug_nrf52>`|PCA10100     |:ref:`nrf52833dk_nrf52820 <nrf52833dk_nrf52820>`|``nrf52833dk_nrf52820``         |
++--------------------------------------------------+-------------+------------------------------------------------+--------------------------------+
+|:ref:`nRF52 DK <ug_nrf52>`                        |PCA10040     |:ref:`nrf52dk_nrf52832 <nrf52dk_nrf52832>`      |``nrf52dk_nrf52832``            |
++--------------------------------------------------+-------------+------------------------------------------------+--------------------------------+
+
+.. set21_end
+
+
+Set used by BLE sample (Peripheral GATT Discovery Manager)
```
Comment (joerchan) [inline doc/nrf/includes/boardname_tables/sample_boardnames.txt:333]: BLE -> Bluetooth LE

Why do we have a different group for this one?

File: doc/nrf/includes/boardname_tables/sample_boardnames.txt [hunk 14]
```diff
@@ -277,3 +278,105 @@ Set used by the SPM sample
...
+|:ref:`nRF52833 DK (emulating nRF52820) <ug_nrf52>`|PCA10100     |:ref:`nrf52833dk_nrf52820 <nrf52833dk_nrf52820>`|``nrf52833dk_nrf52820``         |
++--------------------------------------------------+-------------+------------------------------------------------+--------------------------------+
+|:ref:`nRF52 DK <ug_nrf52>`                        |PCA10040     |:ref:`nrf52dk_nrf52832 <nrf52dk_nrf52832>`      |``nrf52dk_nrf52832``            |
++--------------------------------------------------+-------------+------------------------------------------------+--------------------------------+
+|:ref:`nRF52 DK (emulating nRF52810) <ug_nrf52>`   |PCA10040     |:ref:`nrf52dk_nrf52810 <nrf52dk_nrf52810>`      |``nrf52dk_nrf52810``            |
++--------------------------------------------------+-------------+------------------------------------------------+--------------------------------+
+
+.. set22_end
+
+Set used by BLE sample (Peripheral LBS)
```
Comment (joerchan) [inline doc/nrf/includes/boardname_tables/sample_boardnames.txt:355]: BLE -> Bluetooth LE

Also here a new group for a single sample?
Could we reduce the number of groups?

Comment (umapraseeda) [reply doc/nrf/includes/boardname_tables/sample_boardnames.txt:355]: New group as nRF52840 Dongle is also supported in this sample.

File: doc/nrf/includes/boardname_tables/sample_boardnames.txt (hunk 13 above)
Comment (umapraseeda) [reply doc/nrf/includes/boardname_tables/sample_boardnames.txt:333]: Added a new group as nRF5340 PDK is not there in the supported boards in this sample (comparing this group with set20).

**Comment by umapraseeda** [comment]
Most of the samples are updated and the remaining samples will be updated in separate PRs.

**Pull request closed without merging**